src/main/java/org/opensrp/web/rest/LocationResource.java

```diff
@@ -387,6 +399,48 @@ public void setLocationService(PhysicalLocationService locationService) {
 		return new ResponseEntity<>(identifiers, RestUtils.getJSONUTF8Headers(), HttpStatus.OK);
 
 	}
+	
+	@RequestMapping(value = "/search-location", method = RequestMethod.GET, produces = { MediaType.APPLICATION_JSON_VALUE })
+	@ResponseBody
+	public ResponseEntity<String> searchLocations(@RequestParam(value = "locationTagId", required = false) Long locationTagId,
```

Comment:
Since these params are many and optional just pass `LocationSearchBean`  as the parameter for this method.

Reply:
user HttpServletRequest request